File: release/package/linux/rpm.rb

module Itch
  # RPM package
  def Itch.ci_package_rpm (arch, build_path)
    rpm_arch = to_rpm_arch arch
    gem_dep 'fpm', 'fpm'

    say "Preparing stage2"
    stage2_path = 'rpm-stage'
    prepare_stage2 build_path, stage2_path

    distro_files = ".=/"

    ✓ sh %Q{fpm --force \
      -C #{stage2_path} -s dir -t rpm \
      --rpm-compression xz \
      --name "#{app_name}" \
      --description "#{DESCRIPTION}" \
      --url "https://itch.io/app" \
      --version "#{build_version}" \
      --maintainer "#{MAINTAINER}" \
      --architecture "#{rpm_arch}" \
      --license "MIT" \
      --vendor "itch.io" \
      --category "games" \
      --after-install "release/debian-after-install.sh" \
      -d "p7zip" \
      -d "desktop-file-utils" \
    #{distro_files}
    }

    FileUtils.cp Dir["*.rpm"], "packages/"
  end
end

File: release/package/linux/rpm.js
const $ = require("../../common");
const base = require("./base");

module.exports = {
  packageRpm: async function(arch, buildPath) {
    // RPM package
    const rpmArch = $.toRpmArch(arch);
    $(await $.gemDep("fpm", "fpm"));

    $.say("Preparing stage2");
    const stage2Path = "rpm-stage";
    await base.prepareStage2(buildPath, stage2Path);

    const distroFiles = ".=/";

    $(
      await $.sh(`fpm --force \
      -C ${stage2Path} -s dir -t rpm \
      --rpm-compression xz \
      --name "${$.appName()}" \
      --description "${$.DESCRIPTION}" \
      --url "${$.HOMEPAGE}" \
      --version "${$.buildVersion()}" \
      --maintainer "${$.MAINTAINER}" \
      --architecture "${rpmArch}" \
      --license "MIT" \
      --vendor "itch.io" \
      --category "games" \
      --after-install "release/debian-after-install.sh" \
      -d "desktop-file-utils" \
      -d "libappindicator" \
      -d "libXScrnSaver" \
    ${distroFiles}
    `)
    );

    $(await $.sh("cp *.rpm packages/"));
  },
};
